chain-core/src/state/tendermint.rs

```diff
@@ -37,15 +70,26 @@ impl fmt::Display for TendermintValidatorPubKey {
 }
 
 impl TendermintValidatorPubKey {
+    #[cfg(feature = "base64")]
+    pub fn from_base64(bytes: &[u8]) -> Result<TendermintValidatorPubKey, base64::DecodeError> {
+        let mut result = [0u8; PUBLIC_KEY_SIZE];
+        result.copy_from_slice(&base64::decode(bytes)?);
```

Comment:
copy_from_slice will panic if the lengths don't match, so ideally before this, there should be a check on bytes.len() == PUBLIC_KEY_SIZE